ambry-store/src/main/java/com/github/ambry/store/DiskManager.java

```diff
@@ -235,6 +243,52 @@ void start(boolean shouldRemoveUnexpectedDirs) throws InterruptedException {
     }
   }
 
+  void maybeRecoverBlobStores(int numStoreFailures, Map<PartitionId, Exception> startExceptions) {
+    if (!storeConfig.storeRemoveDirectoryAndRestartBlobStore || numStoreFailures != stores.size()) {
```

Comment:
Why dont we want to attempt restart when only a subset of stores fail?

Reply:
1. It's going to be ok, since recovery means first remove the directory. And then we are left with an empty replica directory to restart. This would be super fast since the majority of the starting time is allocated at reading index files.
2. Changed the logic. The correct logic should be, if all stores fail, we don't want to recover. Since that's a strong indication that the disk has some issues.